client/dashboard/src/pages/catalog/hooks.ts

```diff
@@ -0,0 +1,56 @@
+import { useSdkClient } from "@/contexts/Sdk";
+import { ExternalMCPServer } from "@gram/client/models/components";
+import { queryKeyListMCPCatalog } from "@gram/client/react-query";
+import { useInfiniteQuery } from "@tanstack/react-query";
+import { useEffect, useState } from "react";
+
+interface ServerMeta {
```

Comment:
What do you think about defining this as a zod schema so that we can just try all at once to decode off of the query result and lose some of the nested optional-ness in the downstream code?